python/ql/lib/semmle/python/dataflow/new/internal/TypeTracker.qll

```diff
@@ -188,49 +245,9 @@ module StepSummary {
     smallstepCall(nodeFrom, nodeTo, summary)
   }
 
-  /**
-   * Holds if `nodeFrom` is being written to the `content` of the object in `nodeTo`.
-   *
-   * Note that `nodeTo` will always be a local source node that flows to the place where the content
-   * is written in `basicStoreStep`. This may lead to the flow of information going "back in time"
-   * from the point of view of the execution of the program.
-   *
-   * For instance, if we interpret attribute writes in Python as writing to content with the same
-   * name as the attribute and consider the following snippet
-   *
-   * ```python
-   * def foo(y):
-   *    x = Foo()
-   *    bar(x)
-   *    x.attr = y
-   *    baz(x)
-   *
-   * def bar(x):
-   *    z = x.attr
-   * ```
-   * for the attribute write `x.attr = y`, we will have `content` being the literal string `"attr"`,
-   * `nodeFrom` will be `y`, and `nodeTo` will be the object `Foo()` created on the first line of the
-   * function. This means we will track the fact that `x.attr` can have the type of `y` into the
-   * assignment to `z` inside `bar`, even though this attribute write happens _after_ `bar` is called.
-   */
-  predicate localSourceStoreStep(Node nodeFrom, TypeTrackingNode nodeTo, TypeTrackerContent content) {
-    exists(Node obj | nodeTo.flowsTo(obj) and basicStoreStep(nodeFrom, obj, content))
-  }
+  deprecated predicate localSourceStoreStep = flowsToStoreStep/3;
```

Comment:
I am curious as to why you felt this name change necessary?

Reply:
`localSourceStoreStep` is only used in `smallstepNoCall`, which is now cached, so there is no need to cache this predicate as well. However, it was not a private predicate, so we need to keep it around in case others use it (unlikely).

Reply:
Yes, I was more wondering why you changed the name to `flowsToStoreStep` :-)

Reply:
I had to come up with _some_ name, so I didn't really rename it. `localSourceStoreStep` still exists as a public deprecated predicate.